src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -139,8 +133,12 @@ def _execute(self, all_targets):
   class InvalidShardSpecification(TaskError):
     """Indicates an invalid `--test-shard` option."""
 
+  def _ensure_workdir(self):
```

Comment:
Not for this change, but there really should be something like this in `Task`.

In fact, Task should just always create the workdir.  But again, just musing. Not actionable in this change.